### PR TITLE
Add convex_hull_agg function

### PR DESCRIPTION
**Pull request by yuyongyang800**
convex_hull_agg aggregate function computes convex hull of all input geometries.

### CODE REVIEW

File: presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/aggregation/ConvexHullAgg.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+import static com.facebook.presto.plugin.geospatial.GeometryType.GEOMETRY;
+import static com.facebook.presto.plugin.geospatial.GeometryType.GEOMETRY_TYPE_NAME;
+
+/**
+ * Aggregate form of ST_ConvexHull, which takes a set of geometries and computes the convex hull
+ * of all the geometries in the set. The output is a single geometry.
+ */
+@Description("Returns a geometry that is the convex hull of all the geometries in the set.")
+@AggregationFunction("convex_hull_agg")
+public class ConvexHullAgg
```
[the remark this answers is not in the document]
Comment (yuyongyang800) [reply presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/aggregation/ConvexHullAgg.java]: Code is updated

File: presto-geospatial/src/test/java/com/facebook/presto/plugin/geospatial/aggregation/AbstractAggTestFunctions.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.BiFunction;
+import java.util.stream.Collectors;
+
+import static com.facebook.presto.metadata.FunctionExtractor.extractFunctions;
+import static com.facebook.presto.operator.aggregation.AggregationTestUtils.assertAggregation;
+
+public abstract class AbstractAggTestFunctions
```
[the remark this answers is not in the document]
Comment (yuyongyang800) [reply presto-geospatial/src/test/java/com/facebook/presto/plugin/geospatial/aggregation/AbstractAggTestFunctions.java]: code is updated

File: presto-geospatial/src/test/java/com/facebook/presto/plugin/geospatial/aggregation/AbstractAggTestFunctions.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        extends AbstractTestFunctions
+{
+    private InternalAggregationFunction function;
+
+    protected void register(String funcName)
```
[the remark this answers is not in the document]
Comment (yuyongyang800) [reply presto-geospatial/src/test/java/com/facebook/presto/plugin/geospatial/aggregation/AbstractAggTestFunctions.java]: changed the code based on the comment

**Comment by yuyongyang800** [comment]
I update the sphinx doc to include both aggregate functions.
I think testing convex hull on a large number of geometries may be inefficient. We need to include a file for all the test geometries. Is there such a dataset available? Or perhaps the our end-users could provide one.

File: presto-docs/src/main/sphinx/functions/geospatial.rst [hunk 4]
```diff
@@ -306,6 +306,16 @@ Accessors
 
     Returns the great-circle distance between two points on Earth's surface in kilometers.
 
+Aggregations
+------------
+.. function:: convex_hull_agg(Geometry[]) -> Geometry
+
+    Returns a geometry that is the convex hull of all the geometries in the collection.
+
+.. function:: geometry_union_agg(Geometry[]) -> Geometry
```
[the remark this answers is not in the document]
Comment (yuyongyang800) [reply presto-docs/src/main/sphinx/functions/geospatial.rst]: Removed the doc for geometry_union_agg.

File: presto-docs/src/main/sphinx/functions/geospatial.rst [hunk 5]
```diff
@@ -306,6 +306,16 @@ Accessors
 
     Returns the great-circle distance between two points on Earth's surface in kilometers.
 
+Aggregations
+------------
+.. function:: convex_hull_agg(Geometry[]) -> Geometry
```
[the remark this answers is not in the document]
Comment (yuyongyang800) [reply presto-docs/src/main/sphinx/functions/geospatial.rst]: Updated

File: presto-docs/src/main/sphinx/functions/geospatial.rst [hunk 6]
```diff
@@ -306,6 +306,16 @@ Accessors
...
+
+    Returns a geometry that is the convex hull of all the geometries in the collection.
```
[the remark this answers is not in the document]
Comment (yuyongyang800) [reply presto-docs/src/main/sphinx/functions/geospatial.rst]: Updated

File: presto-docs/src/main/sphinx/functions/geospatial.rst [hunk 7]
```diff
@@ -306,6 +306,14 @@ Accessors
 
     Returns the great-circle distance between two points on Earth's surface in kilometers.
 
+Aggregations
+------------
+.. function:: convex_hull_agg(Geometry) -> Geometry
+
+    Returns a geometry that is the convex hull of all the geometries in the collection/all input geometries.
```
Comment (mbasmanova) [inline presto-docs/src/main/sphinx/functions/geospatial.rst]: This sentence needs to be removed. It contains the same information as the next one.

Comment (yuyongyang800) [reply presto-docs/src/main/sphinx/functions/geospatial.rst]: Done

**Pull request merged**